pkg/syncer/metadatasyncer.go

```diff
@@ -1721,6 +1723,73 @@ func pvcDeleted(obj interface{}, metadataSyncer *metadataSyncInformer) {
 	}
 }
 
+func pvAdded(obj interface{}, metadataSyncer *metadataSyncInformer) {
+	if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorWorkload && isPodVMOnStretchSupervisorFSSEnabled {
+		ctx, log := logger.GetNewContextWithLogger()
+		// Add owner reference of PV to its corresponding CNSVolumeInfo instance, if it doesn't exist.
+		pv, ok := obj.(*v1.PersistentVolume)
+		if pv == nil || !ok {
+			log.Warnf("pvAdded: unrecognized object %+v", obj)
+			return
+		}
+		log.Debugf("pvAdded: PV object %+v", pv)
+		if pv.Spec.CSI == nil {
+			log.Infof("PV %q is not a vSphere CSI volume. Skipping pvAdded functionality.", pv.Name)
+			return
+		}
+		if pv.Status.Phase != v1.VolumeBound {
+			log.Infof("PV %q not bound yet. Skipping to update owner reference "+
+				"on the corresponding CNSVolumeInfo object.", pv.Name)
+			return
+		}
+		err := setOwnerRefForCNSVolumeInfo(ctx, pv)
+		if err != nil {
+			log.Errorf("failed to set owner reference of PV %q on CNSVolumeInfo object %q. Error: %+v",
+				pv.Name, pv.Spec.CSI.VolumeHandle, err)
+			return
+		}
+	}
+}
+
+func setOwnerRefForCNSVolumeInfo(ctx context.Context, pv *v1.PersistentVolume) error {
+	log := logger.GetLogger(ctx)
+	volumeInfoObj, err := volumeInfoService.GetVolumeInfoForVolumeID(ctx, pv.Spec.CSI.VolumeHandle)
+	if err != nil {
+		return logger.LogNewErrorf(log, "failed to fetch CNSVolumeInfo object for volumeID: %q",
+			pv.Spec.CSI.VolumeHandle)
+	}
+	if len(volumeInfoObj.OwnerReferences) > 0 {
```

Comment:
Could we be having more than 1 ownerref in the future on volumeInfoObj?

Reply:
I can't think of an object which will be an owner ref to CNSVolumeInfo other than a PV